Swap to jekyll 2

vuw-ecs-kevin opened the pull request:
These commits take the swcarpentry:gh-pages branch, as it stood after
```
commit a975469c9053fa1ebf3e6a017199c04b7986597d
Merge: c7e0d30 83a83c8
Author: Greg Wilson <gvwilson@third-bit.com>
Date:   Thu May 12 10:37:12 2016 -0400

    Merge branch 'gh-pages' of github.com:swcarpentry/lesson-template into gh-pages
```
and modifies the underlying "lesson template" so that it matches the current lesson
template for lessons that use the Jekyll build environment and which have content
in their own repository and have manually added the styles repository as a remote.

The procedure for moving an old-style (not remote styles repository) lesson to the
new stye, follows the ouline of the link mentioned in:

   http://lists.software-carpentry.org/pipermail/discuss/2017-January/004982.html

although, as some of this work had already been investigated, it's a much smaller
set of diffs than detailed in that link.

vuw-ecs-kevin committed:
Remove top-level HTML files

vuw-ecs-kevin committed:
Move the top level MD files into _episodes

vuw-ecs-kevin committed:
eplace css/ img/ js/ dirs with assets/

vuw-ecs-kevin committed:
Move discussion.md into _extras/
 instructors.md becomes _extras/guide.md

vuw-ecs-kevin committed:
Add in new top level dirs/files. Move tools to bin/tools

vuw-ecs-kevin committed:
Add in new, and modify existing, files to match new lesson template.

vuw-ecs-kevin committed:
Modify the _episode MD files for new template

vuw-ecs-kevin committed:
Modify for changes to lesson-example, since original work

vuw-ecs-kevin committed:
Typo

vuw-ecs-kevin committed:
Git meta-files from the new lesson template
 Note that lesson-example has .gitignore file in
  _episodes and _extras, however we now have content
  in those dirs, so the .gitignore files are not needed.

vuw-ecs-kevin committed:
Remove any "Only in shell-extras" files
 index.md still referred to "lesson_homepage" not just "lesson"

vuw-ecs-kevin committed:
Forgot to generate all_figures for the first time

vuw-ecs-kevin committed:
Update SSH episode to use {: bash}; questions: and keypoints:

vuw-ecs-kevin committed:
Update remaining episodes to use {: bash}

vuw-ecs-kevin committed:
Fixup references to this lessons GiHub repo
 was still using the lesson-example URIs

vuw-ecs-kevin committed:
Adds a "site.index" variable that allows for a fully offline capable
 rendering of the lessons.
This means that no web server is required to navigate the lesson. All
 browsing is done as file:/// links
Furthermore, if a lesson's _config.yml sets the variable to be blank,
 then the URIs within the lesson content in the rendered site will be
 the same as if the site.index variable wasn't there.

vuw-ecs-kevin committed:
Uses the CONTRIBUTING.md, that is already in the repository, to
 create an HTML file within the _site, that an offline rendering
 can then link to and present, rather than having a hanging link
 to a file in a remote GitHub repo's blob, in the footer.
Also fixes up internal links (Licence and Code of Conduct) within
 the CONTRIBUTING.md file, so that they can also link to the HTML
 files that the lesson generates, and not to the MD files.

vuw-ecs-kevin committed:
Make the Citation a page, not a blob file in the repo.

kynan commented on index.md:
```diff
@@ -1,28 +1,26 @@
-5. [Directory structure](05-directory-structure.html)
-6. [Job control](06-job-control.html)
-7. [Aliases and the bashrc](07-aliases.html)
-8. [Environment variables](08-environment-variables.html)
```
Was this intentionally removed?

wking replied:
> Was this intentionally removed?

Probably, because it's now auto-generated:

* `index.md` [uses the `lesson` layout][1].
* The lesson layout [includes the syllabus][2].
* The syllabus is [compiled (by Jekyll) from the episodes][3].

[1]: https://github.com/swcarpentry/lesson-example/blame/dbcd51d135cc2f17b4d870e3af856c8ea7e1a95b/index.md#L2
[2]: https://github.com/swcarpentry/lesson-example/blob/dbcd51d135cc2f17b4d870e3af856c8ea7e1a95b/_layouts/lesson.html#L6
[3]: https://github.com/swcarpentry/lesson-example/blob/dbcd51d135cc2f17b4d870e3af856c8ea7e1a95b/_includes/syllabus.html

vuw-ecs-kevin replied:
As Trevor points out, nothing has been removed. 
This commit,  	f83c399, just puts all of the original content 
into the `_episodes` directory, and after that, the front page
content is automatically generated.
It currently looks like

```
Software Carpentry’s Introduction to the Unix Shell describes the
history of the shell, basic commands, and some of the philosophy
that makes it such a powerful tool. This lesson aims to supplement
the material in that lesson with short topical lessons on particular
shell features that are useful to researchers.

    Prerequisites

    All of this material assumes that learners have
    mastered the basic Unix shell lesson.

    Getting Ready

    A short paragraph describing what learners need
    to do before tackling this lesson.

Other Resources

    Reference
    Discussion
    Instructor’s Guide

 	Setup 			Download files used on the lesson.
00:00 	Manual Pages 	
00:10 	Working Remotely 	How do I use ‘ssh’ and ‘scp’ ?
00:20 	Transferring files 	
00:30 	Permissions 	
00:40 	Directory structure 	
00:45 	Job control 	
00:50 	Aliases and the .bashrc 	
00:50 	Shell Variables 	
01:00 	Finish
```
although I believe, from looking at current lessons, that the
`Other Resources` and `Getting Ready` bits, which I left in
so as not to alter the existing content, would go from that
front page, as they have become items in the Extras menu.

kynan replied:
OK, thanks, didn't know this was auto generated now!

kynan reviewed (approved):
LGTM, apart from a possible issue with `index.md`

vuw-ecs-kevin committed:
Align the "Where to Contribute" info with shell-novice

vuw-ecs-kevin committed:
Merge commit '4b4611e' into fully-offline-capable

vuw-ecs-kevin committed:
Import raw intermediate/shell/04-job.md content

As found at:
  https://github.com/swcarpentry/DEPRECATED-bc/blob/gh-pages/intermediate/shell/04-job.md
Last commit there: a6fd62d on Jun 19, 2014

vuw-ecs-kevin committed:
Update episode to use {: .bash} and {: .output}

Fixup use of glossary.html for reference.html

Make slight wording changes around the reference to this lesson's
 SSH episode.

vuw-ecs-kevin committed:
Make initial attempt to refactor 06-job-control content

vuw-ecs-kevin committed:
Merge commit 'ce1338ab118a0046d449b2afc88e69c913a75c20' into fully-offline-capable

vuw-ecs-kevin committed:
Merge commit '58bb922d67db948a84f0bad645890075318df429' into fully-offline-capable

vuw-ecs-kevin committed:
Merge commit 'd77d709593d7ca85dbaa9363e4a49f63c49b13ee' into fully-offline-capable

vuw-ecs-kevin committed:
Add some questions to the 06-job_control episode

vuw-ecs-kevin committed:
Merge commit '3580d8e' into fully-offline-capable

vuw-ecs-kevin committed:
Fixup the Glossary terms and links

kynan commented on reference.md:
```diff
@@ -23,8 +23,46 @@ to allow other pages to hyperlink to specific glossary entries.)
-key word 1
-:   explanation 1
+remote login
+: FIXME
```
Are you looking to add those as part of this PR?

vuw-ecs-kevin replied:
I should certainly take the left-over bits from the new lesson template's note out,
indeed, apologies for not doing so. I fell victim to an update of the github-pages
Gem not updating the Kramdown Gem, so I wasn't seeing any auto-creation and
assumed it was something specific to the GH pages endpoint. Now that I am aware 
of the version history of the Kramdown Gem, I have seen that it works as expected
when rendered locally.

Reason for leaving the FIXMEs in the Glossary is that, at present, the lessons still
have the original links, (rather, links that  now map to the RefIDs in the Glossary on the
Reference page) so one would want to remove both reference and defintion, if they 
were thought not to be required, or fix the definition in the Glossary if they are.

Given that people are presumably seeing those terms for the "first time", filling in
the defintions is probably the way to go, but for now, the fact they are the target
of links in the lesson suggests they should stay there, as part of the hopefully on-
going ressurection of the content in this lesson.

Interestingly, now that I have had a look at it, the GitHub rendition of that page

https://github.com/vuw-ecs-kevin/shell-extras/blob/swap-to-jekyll-2/reference.md

doesn't appear to be formatting the terms as a DL list ?

vuw-ecs-kevin replied:
> doesn't appear to be formatting the terms as a DL list ?

Because I'm not looking at things on the gh-pages branch!

vuw-ecs-kevin committed:
Remove lesson template guidelines from content

vuw-ecs-kevin committed:
Merge commit 'e09d7fe2d803cecf1918b8563276ea80cdb316f7' into fully-offline-capable

vuw-ecs-kevin committed:
Merge commit 'fd6782caf6f30d44036474fc0bcca963265a32af' into import-intermediate-shell-content

vuw-ecs-kevin committed:
Merge commit 'e09d7fe2d803cecf1918b8563276ea80cdb316f7' into import-intermediate-shell-content

vuw-ecs-kevin committed:
Add offline capability to Glossary terms

vuw-ecs-kevin committed:
Fix typos spotted by kynan

vuw-ecs-kevin committed:
Merge commit '4797fe6a788f750d0eea10742f6eeface695896f' into fully-offline-capable

vuw-ecs-kevin committed:
Fix a loss of space introudced in 9b4bee8b

vuw-ecs-kevin committed:
Merge commit 'f71036d5e144928762b4530a857161c2f1a173a' into fully-offline-capable

vuw-ecs-kevin commented:
Anything likely to happen with this ? Or indeed, with the Shell Extras lessons in general ?

I note that a fair bit of the material covered in the Shell Extras lessons was used in a recent HPC "scaffolding" effort

https://software-carpentry.org/blog/2017/06/hpccarpentry.html

where it's clear that there's a need for people to be exposed to the basic UNIX tools and methodologies, BEFORE they start to think about any actual changes in approaches to coding for HPC resources.

The existing Shell Extras lessons, should they ever be given the chance to become part of the "Carpentry syllabus" by being formally given the current Look&Feel, clearly fill a need, but there's been little activity on here, although KristinaC does get a mention in that HPC "scaffolding" effort piece?

vuw-ecs-kevin committed:
Updates underlying template to match swcarpentry/styles v9.2.4

brandoncurtis commented:
Didn't see this PR and accidentally duplicated this conversion work on another fork.

Is there any additional work needed to get this merged?  I'd like to help fill in the blanks in this lesson for my new-instructor checkout.

vuw-ecs-kevin commented:
I'll probably end up cross-posting between this PR and Brandon's here.
or more likely duplicating this comment, but I wanted to point out that 
what got pulled in by this PR, actually goes a couple of stages further
than the mere "Swap to jekyll 2", in that it also contains (should really 
have been on a seperate branch) some early thoughts I had had about
simple changes to the underlying styles template that would allow for
a correct navigation of the Lesson content, when viewing offline.

These changes don't affect the L&F of this lesson (one of the nice 
things about the extra approach), however, anyone cloning it may 
not be getting exactly what they think they are getting.

If there's a need to back those extras out, then I can look to do that.

FWIW, I decided to develop the "fully offline capable" idea in a standalone
lesson that I have since created here:

https://github.com/vuw-ecs-kevin/offline-capable-lesson

with the rendered version visble here:

https://vuw-ecs-kevin.github.io/offline-capable-lesson/

ChristinaLK commented:
I think it's fine for now -- it looks like Raniere is in the loop, so everything will end up where it needs to eventually.

rgaiacs commented:
> I think it's fine for now -- it looks like Raniere is in the loop, so everything will end up where it needs to eventually.

I would add @ChristinaLK as one of the reviews for https://github.com/swcarpentry/lesson-example/pull/127 but GitHub doesn't allow me. :-( Without her review I'm not sure if everything will end up where it needs to. ;-)
